packages/oui-datagrid/src/datagrid.controller.js

```diff
@@ -0,0 +1,206 @@
+            this.hasActionMenu = true;
+        }
+
+        this.columns = builtColumns.columns;
+
+        if (this.rowsLoader) {
+            this.paging = this.ouiDatagridPaging.createRemote(this.columns, builtColumns.currentSorting, this.pageSize, this.rowLoader, this.rowsLoader);
+            this.refreshData(() => this.paging.setOffset(1));
+        } else {
+            this.paging = this.ouiDatagridPaging.createLocal(this.columns, builtColumns.currentSorting, this.pageSize, this.rowLoader, this.rows);
+
+            if (this.rows) {
+                this.refreshData(() => this.paging.setRows(this.rows));
+            }
+        }
+
+        // Manage responsiveness
+        this.scrollablePanel = this.$element[0].querySelector(".oui-datagrid-responsive-container__overflow-container");
+        if (this.scrollablePanel) {
+            angular.element(this.$window).on("resize", this.checkScroll);
+            angular.element(this.scrollablePanel).on("scroll", this.checkScroll);
+        }
+    }
+
+    $doCheck () {
+        if (!angular.equals(this.previousRows, this.rows)) {
+            this.previousRows = angular.copy(this.rows);
+
+            if (this.rows && this.paging) {
+                this.refreshData(() => this.paging.setRows(this.rows));
+            }
+        }
+    }
+
+    $onDestroy () {
+        angular.element(this.$window).off("resize", this.checkScroll);
+        angular.element(this.scrollablePanel).off("scroll");
+    }
+
+    getParentScope () {
+        return this.$scope.$parent;
+    }
+
+    hasProperty (obj, prop) { // eslint-disable-line
+        if (!obj) {
+            return false;
+        }
+
+        return hasProperty(obj, prop);
+    }
+
+    // TODO
+    handleError () { // eslint-disable-line
+        // do nothing.
+    }
+
+    onPaginationChange ($event) {
+        this.refreshData(() => {
+            this.paging.setOffset($event.offset);
+            this.paging.setPageSize($event.pageSize);
+        }, true);
+    }
+
+    scrollToTop () {
+        // Small delay otherwise rows could not be rendered
+        // yet and position would be wrong
+        this.$timeout(() => {
+            this.$element[0].scrollIntoView(true);
+        });
+    }
+
+    refreshData (callback, skipSort, requireScrollToTop) {
+        if (this.loading) {
+            return this.$q.reject(false);
+        }
+
+        this.loading = true;
+
+        return this.$q.when(callback())
+            .then(() => {
+                this.displayedRows = DatagridController.createEmptyRows(25); // eslint-disable-line no-magic-numbers
+                return this.paging.loadData(skipSort);
+            })
+            .then(result => {
+                this.displayedRows = result.data;
+                if (requireScrollToTop) {
+                    this.scrollToTop();
+                }
+                setTimeout(() => this.checkScroll(), 1000); // eslint-disable-line no-magic-numbers
```

Comment:
Why 1000? Could you add a comment

Reply:
Moved into a const and added a comment to explain why.